pkg/cloudprovider/types.go

```diff
@@ -261,6 +293,16 @@ type Offering struct {
 	Price               float64
 	Available           bool
 	ReservationCapacity int
+
+	overlayApplied bool
+}
+
+func (o *Offering) ApplyOverlay() {
+	o.overlayApplied = true
+}
+
+func (o *Offering) IsAnOverlayApplied() bool {
```

Comment:
nit: naming, similar suggestion applies elsewhere
```suggestion
func (o *Offering) IsOverlaid() bool {
```